src/librustc_mir/transform/const_prop.rs

```diff
@@ -410,15 +410,10 @@ impl<'mir, 'tcx> ConstPropagator<'mir, 'tcx> {
         }
     }
 
-    fn use_ecx<F, T>(&mut self, source_info: SourceInfo, f: F) -> Option<T>
+    fn use_ecx<F, T>(&mut self, f: F) -> Option<T>
     where
         F: FnOnce(&mut Self) -> InterpResult<'tcx, T>,
     {
-        self.ecx.tcx.span = source_info.span;
-        // FIXME(eddyb) move this to the `Panic(_)` error case, so that
```

Comment:
@eddyb please check that it is correct to remove this FIXME now -- the short-circuiting is gone.

Reply:
Yupp, the important part is always calling `f`, i.e. as if it were linear.